packages/abi/src/gen/renderers/ts/typers/iterators.ts

```diff
@@ -0,0 +1,38 @@
+import { ARRAY_REGEX } from '../../../../matchers/sway-type-matchers';
+
+import { mapComponents } from './struct';
+import type { Typer } from './types';
+
+export const tupleTyper: Typer = ({ abiType }, typer) =>
+  mapComponents({ parent: abiType, includeComponentNames: false, typer });
+
+export const arrayTyper: Typer = ({ abiType }, typer) => {
+  const length = ARRAY_REGEX.exec(abiType.swayType)?.[2];
+
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const { type } = abiType.components![0]!;
+  const mapped = typer({ abiType: type, asReference: true });
+
+  const input = `ArrayOfLength<${mapped.input}, ${length}>`;
+  const output = `ArrayOfLength<${mapped.output}, ${length}>`;
+
+  return {
+    input,
+    output,
+    fuelsTypeImports: mapped.fuelsTypeImports,
+    commonTypeImports: ['ArrayOfLength', ...(mapped.commonTypeImports ?? [])],
+  };
+};
+
+export const vectorTyper: Typer = ({ abiType }, typer) => {
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const { type } = abiType.components![0]!;
+  const mapped = typer({ abiType: type, asReference: true });
+  const input = `${mapped.input}[]`;
+  const output = `${mapped.output}[]`;
```

Comment:
> I was a fan of the `Vec` type

I can see the point in signalling the underlying type with `Vec`, although I prefer denoting it with the `[]` as it implies the same thing. Let's see what the rest of the team says. I can also create a `Tuple<T1, T2, T3>` element in the same vein instead of `[T1, T2, T3]` although, again, I prefer the `[T1, T2, T3]` approach.